content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV_bounded[3];  // Will be bound to t1 register... Until t3 (inclusive).
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t4 register, and will take over t5+.
+        RWTexture2D<float4>      m_texUAV_bounded[5];  // Will be bound to u0 register... Until u4 (inclusive).
+    	RWTexture2D<float4>      m_texUAV_unbounded[]; // Will be bound to u5 register, and will take over u6+.
+        Sampler                  m_sampler1_bounded[7]; // Will be bound to s0 register... Until s6 (inclusive).
+        Sampler                  m_sampler2_bounded[3]; // Will be bound to s7 register... Until s9 (inclusive).
+        Sampler                  m_sampler_unbounded[]; // Will be bound to s10 register, and will take over s11+.
+        ConstantBuffer<MyStruct> m_cb_array[2]; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register)... Until     b2 (inclusive).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b3 register.
+        ConstantBuffer<MyStruct> m_cb_unbounded[]; // Will be bound to b4 register, and will take over b5+.
+    };
+```
+Now, let's review what will happen to all the examples above when using '--unique-idx', which is the case for Vulkan & Metal.
+
+### Example 6: Based on Example 1, with '--unique-idx'
+Same code as Example 1, but it is assumed to be compiled with '--unique-idx' (ON), which is required by Vulkan & Metal. Please note how the register index always increments, regardless of the resource type.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t2 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u3 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u4 register.
+        Sampler                  m_sampler1;     // Will be bound to s5 register.
+        Sampler                  m_sampler2;     // Will be bound to s6 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b7 register.
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b8 register.
+    };
```

Comment:
```suggestion  
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

//For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
// The ConstantBuffer reserved for SRG1 binds to b0 register.
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV1;     // Binds to t1 register.
    Texture2D<float4>        m_texSRV2;     // Binds to t2 register.
    RWTexture2D<float4>      m_texUAV1;     // Binds to u3 register.
    RWTexture2D<float4>      m_texUAV2;     // Binds to u4 register.
    Sampler                  m_sampler1;    // Binds to s5 register.
    Sampler                  m_sampler2;    // Binds to s6 register.
    ConstantBuffer<MyStruct> m_cb1;         // Binds to b7 register.
    ConstantBuffer<MyStruct> m_cb2;         // Binds to b8 register.
};
```
* Moved comment: "(Because the ConstantBuffer reserved for….)" to simplify the comments for each resource declaration. 
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

Reply:
Fixed.